src/Spice86.Core/Emulator/CPU/CfgCpu/InstructionRenderer/InstructionRendererHelper.cs

```diff
@@ -0,0 +1,318 @@
+        return _registerRenderer.ToStringSegmentRegister(registerIndex);
+    }
+
+    public string ToString(InstructionField<byte> field) {
+        if (field.UseValue) {
+            return ToHex(field.Value);
+        }
+
+        return ToAbsolutePointer(8, field.PhysicalAddress);
+    }
+
+    public string ToString(InstructionField<ushort> field) {
+        if (field.UseValue) {
+            return ToHex(field.Value);
+        }
+
+        return ToAbsolutePointer(16, field.PhysicalAddress);
+    }
+
+    public string ToString(InstructionField<uint> field) {
+        if (field.UseValue) {
+            return ToHex(field.Value);
+        }
+
+        return ToAbsolutePointer(32, field.PhysicalAddress);
+    }
+
+    public string ToString(InstructionField<sbyte> field) {
+        if (field.UseValue) {
+            return ToHex(field.Value);
+        }
+
+        return ToAbsolutePointer(8, field.PhysicalAddress);
+    }
+
+    public string ToString(InstructionField<short> field) {
+        if (field.UseValue) {
+            return ToHex(field.Value);
+        }
+
+        return ToAbsolutePointer(16, field.PhysicalAddress);
+    }
+
+    public string ToString(InstructionField<int> field) {
+        if (field.UseValue) {
+            return ToHex(field.Value);
+        }
+
+        return ToAbsolutePointer(32, field.PhysicalAddress);
+    }
+
+    public string ToString(InstructionField<SegmentedAddress> field) {
+        if (field.UseValue) {
+            return ToHex(field.Value);
+        }
+
+        return ToAbsolutePointer(32, field.PhysicalAddress);
+    }
+
+    public string ToStringRm(int targetBitWidth, ModRmContext modRmContext) {
+        if (modRmContext.MemoryAddressType == MemoryAddressType.NONE) {
+            // then it's a register
+            return ToStringR(targetBitWidth, modRmContext);
+        }
+        return ToStringMemoryAddress(targetBitWidth, modRmContext);
+    }
+
+    public string ToStringR(int bitWidth, ModRmContext modRmContext) {
+        return ToStringRegister(modRmContext.RegisterMemoryIndex, bitWidth);
+    }
+
+    public string ToStringMemoryAddress(int targetBitWidth, ModRmContext modRmContext) {
+        if (modRmContext.MemoryAddressType == MemoryAddressType.NONE) {
+            return "";
+        }
+
+        if (modRmContext.SegmentIndex == null) {
+            throw new ArgumentException("SegmentIndex is null");
+        }
+        string segment = ToStringSegmentRegister(modRmContext.SegmentIndex.Value);
+        string offset = ToStringModRmMemoryOffset(modRmContext);
+        return ToSegmentedPointer(targetBitWidth, segment, offset);
+    }
+
+    public string ToStringModRmMemoryOffset(ModRmContext modRmContext) {
+        if (modRmContext.MemoryOffsetType == MemoryOffsetType.NONE) {
+            return "";
+        }
+
+        string displacement = ToStringModRmDisplacement(modRmContext);
+        string join = "+";
+        if (displacement.StartsWith("-")) {
```

Comment:
This might use a regex instead ?

There are example of compile time generated regex in the codebase. Just an idea.